siliconcompiler/core.py

```diff
@@ -4140,10 +4140,32 @@ def run(self):
                     # TODO: breakpoint logic:
                     # if task is breakpoint, then don't launch while len(running_tasks) > 0
 
+                    successful_deps = []
+                    deps_full_len = len(deps)
+                    tool, _ = self._get_tool_task(
+                        processes[task]._args[0],
+                        processes[task]._args[1])
```

Comment:
nit: I'm slightly uncomfortable with relying on these private members, it might be worth stashing the step, index tuple in a separate dictionary (or perhaps refactoring to use `(step, index)` as the actual key for all these dictionaries, if that makes sense).

Happy to keep for now and save the refactor for a separate PR, though.

Reply:
@gadfort @WRansohoff 
Just started refactoring but then I realised at some points the status dict does not use the `f'{step, index}'` format.
Most of these are from the server using it.
Examples I found: `status['remote_cfg']` , `status['jobhash']`, `status['slurm_constraint']`, `chip.status['slurm_account']`, `chip.status['max_fs_bytes']`, `chip.status['watchdog']`
Mixed key types in the dictionary are probably not that good of an idea.
Do you think for those setting index to 0 would make sense? i.e.`status[('jobhash', '0')]`
Or should we just completly move these out of the status dict?

In either case I think a separate PR makes more sense...

Reply:
SC-585

Reply:
okay, I think once we record the success or failure of a run in the schema, this status dict is not needed any longer.